src/main/java/com/datalogics/pdf/samples/manipulation/RemoveInteractivity.java

```diff
@@ -127,6 +128,10 @@ public static void removeInteractivity(final URL inputUrl, final URL outputUrl)
         try {
             pdfDoc = DocumentUtils.openPdfDocument(inputUrl);
 
+            // Before removing any interactive elements, generate their appearances using the AppearanceService
+            // to ensure that the document has all required appearances so the document looks correct at the end.
+            AppearanceService.generateAppearances(pdfDoc, null, null);
```

Comment:
Since you have an apContext object; I'm curious why you don't pass it in as a parameter here.